charts/gateway-helm/templates/certgen-rbac.yaml

```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: {{ include "eg.fullname" . }}-certgen
-  namespace: '{{ .Release.Namespace }}'
+  namespace:  '{{ .Values.namespace | default .Release.Namespace }}'
```

Comment:
is this recommand way for helm by introuducing a namespace field in `values.yaml`?

Reply:
Yes, in values.yaml will be only "namespace: name-of-ns" and it's easy to configure in whole chart. 
Because if you use Umbrella templating .Release.Namespace is always default and it's not possible to change it.

Reply:
`helm template -n ` didn't work?

Reply:
`helm template -n` you can't use if using Umbrella chart, you need to define namespace for every child Helm chart.

Reply:
I didn't think it's a good practice to add things like this.

Reply:
Why? 
For example `ingress-nginx` has option `namespaceOverride` (this is more complex, than my PR):

https://github.com/kubernetes/ingress-nginx/pull/10539/files

Reply:
curious why a  `namespaceOverride` field is needed when it can be achieved in helm natively ?
```
helm install eg-marketing oci://docker.io/envoyproxy/gateway-helm --version v1.1.0 -n marketing --create-namespace
```

Reply:
because if you managing plenty of helm charts over [umbrella approach](https://helm.sh/docs/howto/charts_tips_and_tricks/#complex-charts-with-many-dependencies), you have to define namespace for every subchart.

Reply:
IMO, we shouldn't support this(the benefits of introduction are far smaller than the complications), you can do this in your own fork.

Reply:
Which are the complications you see? Nothing changes in chart logic, only one option is added. 
Many CNCF projects adopted this logic.

Reply:
Another example in huge project which ArgoCD is:

https://github.com/argoproj/argo-helm/blob/fd70c609bd6e98e8b067c7c4915957c51acd786c/charts/argo-cd/templates/_helpers.tpl#L242

https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/values.yaml#L11